File: pset6/mario/mario.py
import cs50


def main():

    # prompt the user for positive integer not greater than 8
    while True:
        height = cs50.get_int("Height: ")
        if height > 0 and height <= 8:
            break
    # print pyramid
    print_pyramid(height, -1)


def print_pyramid(height, row):
    # base case
    if height == 0:
        return
    # count the number of spaces
    row += 1

    # recursivally print another smaller row
    print_pyramid(height - 1, row)

    # print spaces
    print_spaces(row)

    # print left pyramid
    for i in range(height):
        print('#', end='')

    # print gap in the middle
    print_spaces(2)

    # print the right pyramid
    for i in range(height):
        print('#', end='')

    # gow to the new line
    print('\n', end='')


# function for printing explicit amount of spaces
def print_spaces(n):
    for i in range(n):
        print(' ', end='')


if __name__ == "__main__":
    main()

File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // variables
    int height, i, j, k, m;
    do
    {
        height = get_int("Height: \n");
    }
    while ((height < 1) || (height > 8));


    // printing bricks
    // controlling row number by i
    for (i = 1; i < height + 1; i++)
    {
        // empty spaces in a row
        for (j = 0; j < height - i; j++)
        {
            printf(" ");
        }
        // left bricks in a row
        for (k = 0; k < i; k++)
        {
            printf("#");
        }
        // gap
        printf("  ");
        // right bricks
        for (m = 0; m < i; m++)
        {
            printf("#");
        }
        // new row
        printf("\n");
    }
}